### PR TITLE
Respect `develop` when installing from git

### LINKED ISSUE #1080: Poetry does not respect the `develop = false` option when installing from git/http
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Ubuntu Linux 19.04
- **Poetry version**: 0.12.15
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

## Issue

Related to but different from: https://github.com/sdispater/poetry/issues/1073

We have a private library we install from Github, this library was written with the aid of Poetry and as such has _no_ `setup.py`. 

Installation fails when pip 19.1 is installed due to Poetry attempting to install it in _editable_ mode. 

In this instance editable mode is not desired or needed. I noted that for directory based installed you can specify `develop = false` to have poetry install in a non-editable mode. But this does not appear to be an option when installing from git/http.

### PR DESCRIPTION
**Pull request by fantix**
* fixes #1080, again

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by lambdalisue** [comment]
So why this PR has not been reviewed or merged for a month? I've been facing [an issue](https://github.com/python-poetry/poetry/issues/2276) which is not resolved without this PR so if any help is required, I'm really welcome to do.

File: poetry/puzzle/provider.py [hunk 1]
```diff
@@ -169,6 +169,7 @@ def search_for_vcs(self, dependency):  # type: (VCSDependency) -> List[Package]
             dependency.source,
             dependency.reference,
             name=dependency.name,
+            develop=dependency.develop,
```
Comment (sdispater) [inline poetry/puzzle/provider.py:172]: This is not the right place to put this, `get_package_from_vcs` should not have any notion of `develop`.

So, similarly to what is done in `search_for_directory`, we can do:

```python
package.develop = dependency.develop
```

File: poetry/puzzle/provider.py [hunk 2]
```diff
@@ -182,8 +183,8 @@ def search_for_vcs(self, dependency):  # type: (VCSDependency) -> List[Package]
 
     @classmethod
     def get_package_from_vcs(
-        cls, vcs, url, reference=None, name=None
-    ):  # type: (str, str, Optional[str], Optional[str]) -> Package
+        cls, vcs, url, reference=None, name=None, develop=True,
+    ):  # type: (str, str, Optional[str], Optional[str], Optional[bool]) -> Package
```
Comment (sdispater) [inline poetry/puzzle/provider.py:187]: See my comment above.

File: poetry/puzzle/provider.py [hunk 3]
```diff
@@ -206,6 +207,7 @@ def get_package_from_vcs(
             package.source_type = "git"
             package.source_url = url
             package.source_reference = revision
+            package.develop = develop
```
Comment (sdispater) [inline poetry/puzzle/provider.py:210]: See my comment above.

**Comment by neersighted** [comment]
Closing this as this is long stale and the issue has been solved in master/Poetry 1.2 beta for a long time. Please open a new issue/PR if you still observe broken behavior.

**Comment by tomoyaY** [comment]
@neersighted which commits the issue was addressed?

**Comment by neersighted** [comment]
https://github.com/python-poetry/poetry/commit/f0855d984d9b0b38657c8b8522d3cfa36280b47c
and https://github.com/python-poetry/poetry-core/commit/fc4ae2d53c7ffc877ad50028a6ec132951dfda58

If you are having trouble with this functionality, please open a new Discussion or join Github. If you think you are experiencing a bug with it, please open an issue with steps for reproduction/your project files.

**Comment by tomoyaY** [comment]
Thank you for your quick reply!!

**Pull request closed without merging**